### PR TITLE
fix: validate CLI --port / --hostname values, reject missing/flags-as-values

### LINKED ISSUE #1012: vinext dev/start CLI arg parser silently accepts missing values for --port / --hostname (NaN, undefined, swallows next flag)
`vinext dev` and `vinext start` accept `--port` / `--hostname` (and their short aliases `-p` / `-H`) without validating that the next argv slot actually contains a value. The parser blindly consumes `args[++i]`, so:

1. `vinext dev --port` (no value) → `port = NaN` → server tries to `listen(NaN)` and either crashes or binds an OS-assigned random port silently.
2. `vinext dev --hostname` (no value) → `hostname = undefined`.
3. `vinext dev --port --hostname 0.0.0.0` (typo: missing port value) → `--hostname` is consumed as the port value, parses to `NaN`, AND the hostname flag is lost. User loses BOTH settings with no warning.

## Reproduce

The bug lives in the CLI parser at `dist/cli.js:71-73`. Same pattern in source (`src/cli.ts`):

```js
} else if (arg === "--port" || arg === "-p") result.port = parseInt(args[++i], 10);
  else if (arg.startsWith("--port=")) result.port = parseInt(arg.split("=")[1], 10);
  else if (arg === "--hostname" || arg === "-H") result.hostname = args[++i];
  else if (arg.startsWith("--hostname=")) result.hostname = arg.split("=")[1];
```

Standalone repro of the parser logic (no need to install vinext):

```js
function parseArgs(args) {
  const result = {};
  for (let i = 0; i < args.length; i++) {
    const arg = args[i];
    if (arg === "--port" || arg === "-p") result.port = parseInt(args[++i], 10);
    else if (arg === "--hostname" || arg === "-H") result.hostname = args[++i];
  }
  return result;
}

console.log(parseArgs(["--port"]));
// → { port: NaN }

console.log(parseArgs(["--hostname"]));
// → { hostname: undefined }

console.log(parseArgs(["--port", "--hostname", "0.0.0.0"]));
// → { port: NaN }   ← --hostname value is silently swallowed too
```

End-to-end repro:

```bash
$ npx vinext@0.0.45 dev --port
# server starts; subsequent fetch on the expected port fails / hits a random port
```

## Why it's a bug

Three concrete failure modes:

1. **Listen on `NaN`** — Node's `server.listen(NaN)` falls back to a random ephemeral port. The user typed `--port` expecting a deterministic port; they get one assigned by the OS. The dev URL printed is correct but the user's bookmarks / proxy configs / wrangler bindings break.
2. **`hostname = undefined`** — Vite's `server.host` falls back to its default (`localhost`), which means the dev server is unreachable from anything but the loopback interface. Common bug shape inside containers / WSL — user explicitly passes `--hostname 0.0.0.0` to fix it, mistypes (case 3 above), and gets the broken default back without a warning.
3. **Argument-eating typo** — `vinext dev --port --hostname 0.0.0.0` parses as `{ port: NaN }` (no hostname). Everything downstream of port-resolution fails, but the error message says "could not bind port NaN" rather than "you forgot the port value."

## Proposed fix

Validate that `args[i + 1]` exists and isn't another flag before consuming it. Suggested shape:

```js
function takeValue(name, args, i) {
  const next = args[i + 1];
  if (next === undefined || next.startsWith("-")) {
    throw new Error(`${name} requires a value (got: ${next ?? "<end of args>"})`);
  }
  return next;
}

// ...
else if (arg === "--port" || arg === "-p") {
  const v = takeValue(arg, args, i);
  i++;
  result.port = parseInt(v, 10);
  if (Number.isNaN(result.port)) throw new Error(`${arg} expects an integer (got: ${v})`);
}
else if (arg === "--hostname" || arg === "-H") {
  result.hostname = takeValue(arg, args, i);
  i++;
}
```

Same shape for any other future flag taking a value.

## Environment

```
vinext: 0.0.45
node: 24
bun: 1.3.13
host: macOS arm64 (also reproduces on linux)
```

The bug is in the parser itself — environment-independent.

### PR DESCRIPTION
**Pull request by NathanDrake2406**
Fixes #1012

## Summary
The vinext CLI argument parser silently consumed missing values for --port / -p and --hostname / -H flags, causing:
1. NaN port → random ephemeral bind
2. Undefined hostname → fallback to localhost
3. Flag-eating typos where --hostname is consumed as a port value

This change adds validation that throws clear, actionable errors.

### What changed
- Extracted parseArgs to cli-args.ts module for testability
- Added takeValue() validation rejecting missing values and flags-as-values
- Added integer validation for --port
- Moved --precompress env var side effect into buildApp call site
- 33 unit tests in tests/cli-args.test.ts

### Behavior parity
Next.js CLI uses similar validation via [next/src/cli/next-dev.ts](https://github.com/vercel/next.js/blob/canary/packages/next/src/cli/next-dev.ts). Error messages follow the standard flag-requires-a-value convention.

## Test plan
33 tests covering: missing value at end of args, flag consumed as value, non-numeric port, long/short forms, =value forms, boolean flags, combined flags, edge cases.

### CODE REVIEW

File: packages/vinext/src/cli-args.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+  verbose?: boolean;
+  turbopack?: boolean;
+  experimental?: boolean;
+  prerenderAll?: boolean;
+  precompress?: boolean;
+};
+
+// Matches long flags (--foo) and single-letter short flags (-x), but not
+// negative numbers (-1) or multi-character non-flag strings starting with -.
+const FLAG_PATTERN = /^(?:--|-[a-zA-Z]$)/;
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/cli-args.ts:22]: Updated — comment now says "Digits and multi-char sequences (e.g. -1, -abc) are excluded by [a-zA-Z] and $."

File: packages/vinext/src/cli-args.ts [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+      default: {
+        // Handle --flag=value forms (e.g. --port=3000, --hostname=0.0.0.0).
+        const eqRaw = tryEqualsForm(arg, "port");
+        if (eqRaw !== null) {
+          result.port = parsePort(eqRaw, "--port");
+          break;
+        }
+        const hostRaw = tryEqualsForm(arg, "hostname");
+        if (hostRaw !== null) {
+          if (hostRaw === "") {
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/cli-args.ts:136]: Left the `=== ""` check — `!hostRaw` would also reject `"0"` as a falsy value. Extracting an `isNonEmpty` helper would be cleaner but the inlined check is fine at this scale (one call site for each flag form). Open to extracting if/when a third value-taking flag is added.

File: packages/vinext/src/cli-args.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+  }
+  return parsed;
+}
+
+/**
+ * Parse CLI arguments into a structured object.
+ *
+ * Handles both `--flag value` and `--flag=value` forms for value-taking flags.
+ */
+export function parseArgs(args: string[]): ParsedArgs {
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply packages/vinext/src/cli-args.ts:81]: Done — added JSDoc noting that `parseArgs` covers dev/build/start/lint/check/init, while deploy uses `parseDeployArgs` with `node:util`.

File: tests/cli-args.test.ts [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+    expect(parseArgs(["--port", "3000", "--port", "4000"])).toMatchObject({ port: 4000 });
+  });
+
+  it("skips unrecognized flags", () => {
+    const result = parseArgs(["--unknown", "value"]);
+    expect(result).not.toHaveProperty("unknown");
+    expect(result).not.toHaveProperty("port");
+    expect(result).not.toHaveProperty("hostname");
+  });
+});
```
[the remark this answers is not in the document]
Comment (NathanDrake2406) [reply tests/cli-args.test.ts:247]: Good catch. Fixed in `takeValue` by rejecting `next === ""` alongside `next === undefined` — this catches empty strings for all flags, not just hostname. Added tests for both `--hostname ""` and `-H ""`.
